extensions/container-image/container-image-docker-common/deployment/src/main/java/io/quarkus/container/image/docker/common/deployment/RedHatOpenJDKRuntimeBaseProvider.java

```diff
@@ -9,10 +9,11 @@
 import java.util.stream.Stream;
 
 /**
- * Can extract information from Dockerfile that uses {@code registry.access.redhat.com/ubi8/openjdk-$d-runtime:$d.$d} as the
+ * Can extract information from Dockerfile that uses {@code registry.access.redhat.com/ubi[89]/openjdk-$d-runtime:$d.$d} as the
  * base image
  */
 class RedHatOpenJDKRuntimeBaseProvider implements DockerFileBaseInformationProvider {
+    private static final Pattern PATTERN = Pattern.compile(".*ubi[89]/openjdk-(\\w+)-runtime.*");
```

Comment:
Maybe leave it ready for ubi10 and above too?

Reply:
Now why didn't I think of that? :)

Reply:
Yeah, I agree it makes sense to be a bit more permissive. I would go with `ubi[0-9]+`.

Worst case if we need to handle version-specific cases, we could do it later.

Reply:
How about this?

```regex
.*ubi([8-9]|[1-9][0-9]+)/openjdk-(\w+)-runtime.*
```

This would match anything >= `ubi8`

Reply:
As you wish but I'm not sure it's worth making it too complex. Basically you just want to catch numbers and that will be good enough.
I won't fight it if you absolutely want this version but I think it makes it hard to maintain for a very narrow source of problem.

Reply:
The part I don't like with `ubi[0-9]+` is that it'll match ubi versions that don't currently exist (anything < 7). Yes ubi7 exists, but probably shouldn't be used.

I could make it a bit simpler and forget about matching the java version?

```regex
.*ubi([8-9]|[1-9][0-9]+)/openjdk-(\w+)-runtime.*
```

Would that be a better compromise?

Reply:
You don't need the `[0-9]` part as the second part will match 10 and above:
```
.*ubi([8-9]|[1-9])/openjdk-(\w+)-runtime.*
```

Reply:
But that will also match things like `ubi2`, which is what i'm trying to avoid. `[8-9]|[1-9][0-9]+` matches any number >= 8, even double digit.

Reply:
> But that will also match things like `ubi2`,

I don't think so, see https://regex101.com/r/xSj9la/1

Reply:
Ahh you didn't have the extra `\d+` in it when i tested it :) `\d+` == `[0-9]+`. I've read in some places that `[0-9]` is preferred to `'\d` because it is more universal, although I'm certainly no regex expert.

Reply:
Hm, maybe you're right about the syntax. I'm fine with whatever works in this case 😀

Reply:
I'm all set now. Just pushed last set of changes.

Reply:
Thanks for the review!

Reply:
FWIW, I agree what you ended up with is a good compromise.